### PR TITLE
chore(deps): update dependency husky to v7

### PR DESCRIPTION
**Pull request by chimurai**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes #13345
| Patch: Bug Fix?          | no
| Major: Breaking Change?  | no
| Minor: New Feature?      | no
| Tests Added + Pass?      | -
| Documentation PR Link    | https://github.com/babel/babel/pull/13345
| Any Dependency Changes?  | dev husky@^7.0.4
| License                  | MIT

Upgrade husky from v3 to v7.

Since v7 has 0 dependencies; This will also removes some transitive (obsolete) dependencies.

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/14011"></a>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
   "private": true,
   "type": "commonjs",
   "scripts": {
+    "prepare": "husky install",
```
Comment (chimurai) [inline package.json:7]: Why husky doesn't autoinstall anymore: https://blog.typicode.com/husky-git-hooks-autoinstall

**Review by nicolo-ribaudo — APPROVED** [review]
Thanks!

File: package.json (hunk 1 above)
Comment (nicolo-ribaudo) [reply package.json:7]: It looks like we just need `postinstall`: https://typicode.github.io/husky/#/?id=yarn-2

**Comment by chimurai** [comment]
Changed `prepare` to `postinstall` 👍

**Review by nicolo-ribaudo — APPROVED** [review]

**Review by existentialism — APPROVED** [review]

**Pull request merged**